x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -213,6 +214,7 @@ protected Object unwrapMultiValue(Object values) {
     protected boolean isFromDocValuesOnly(DataType dataType) {
         return dataType == KEYWORD // because of ignore_above.
                     || dataType == DATETIME
+                    || dataType == CONSTANT_KEYWORD // because a non-existent value is considered the constant value itself
```

Comment:
Could you explain please, I don't get it.

Reply:
We cannot extract the value of a constant_keyword from _source because if there is no source for the field, Elasticsearch will still consider the field as having the constant value.
Take a look at the example in [our docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/constant-keyword.html). No value at indexing time means `level` is `debug`. So, for the second document in the sample, there is no _source for `level`, but the value is still `debug` in case there is a query filtering on that field.